src/core/SparseMatrix.cpp

```diff
@@ -68,6 +68,15 @@ SparseMatrix::SetZeros(){
     fArmaMat = arma::sp_mat(fNCols, fNRows);
 }
 
+void
+SparseMatrix::SetIdentity(){
+    if(!fNRows && !fNCols && fNCols!=fNRows )
```

Comment:
You're right I need to remove the negation. Basically I want to make sure it is a square matrix.